File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailType38AdminService.java
package ma.zs.univ.service.facade.admin.NotificationRetardDeuxiemeNiv;

import java.util.List;
import ma.zs.univ.bean.core.NotificationRetardDeuxiemeNiv.NotificationRetardDeuxiemeNiveauDetailType38;
import ma.zs.univ.dao.criteria.core.NotificationRetardDeuxiemeNiv.NotificationRetardDeuxiemeNiveauDetailType38Criteria;
import ma.zs.univ.zynerator.service.IService;



public interface NotificationRetardDeuxiemeNiveauDetailType38AdminService extends  IService<NotificationRetardDeuxiemeNiveauDetailType38,NotificationRetardDeuxiemeNiveauDetailType38Criteria>  {

    List<NotificationRetardDeuxiemeNiveauDetailType38> findByTypeLocale38Id(Long id);
    int deleteByTypeLocale38Id(Long id);
    long countByTypeLocale38Code(String code);




}

File: code/frontend/src/app/shared/service/admin/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailType38Admin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {NotificationRetardDeuxiemeNiveauDetailType38Dto} from 'src/app/shared/model/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailType38.model';
import {NotificationRetardDeuxiemeNiveauDetailType38Criteria} from 'src/app/shared/criteria/NotificationRetardDeuxiemeNiv/NotificationRetardDeuxiemeNiveauDetailType38Criteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class NotificationRetardDeuxiemeNiveauDetailType38AdminService extends AbstractService<NotificationRetardDeuxiemeNiveauDetailType38Dto, NotificationRetardDeuxiemeNiveauDetailType38Criteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): NotificationRetardDeuxiemeNiveauDetailType38Dto {
        return new NotificationRetardDeuxiemeNiveauDetailType38Dto();
    }

    public constrcutCriteria(): NotificationRetardDeuxiemeNiveauDetailType38Criteria {
        return new NotificationRetardDeuxiemeNiveauDetailType38Criteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/notificationRetardDeuxiemeNiveauDetailType38/';
    }
}
